episodes/threads-and-processes.md

```diff
@@ -81,26 +83,28 @@ def calc_pi_nogil(N):
 ```
 
 The `nopython` argument forces Numba to compile the code without referencing any Python objects,
-while the `nogil` argument enables lifting the GIL during the execution of the function.
+while the `nogil` argument disables the GIL during the execution of the function.
 
 :::callout
 ## Use `nopython=True` or `@numba.njit`
-It's generally a good idea to use `nopython=True` with `@numba.jit` to make sure the entire
+It is generally good practice to use `nopython=True` with `@numba.jit` to make sure the entire
```

Comment:
Actually, the default behaviour has changed so that `nopython=True` is no longer needed. We should update the material to reflect this.

Reply:
I have noticed that. Since I did not start from the environment requested for the workshop (I has missed the direction or it was were added meanwhile), I have worked with Python 3.8 and with the old Numba default.

I received the deprecation warning already in Episode 3: 

> /tmp/ipykernel_16267/2154243854.py:4: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.

So I agree factually. First, however, I would make sure that we manage to have all learners work on the same Python version. I am cautious as to the fact that the yml environment delivers seamlessly for everyone (hint from the experience with the course in Image Processing).

So, I leave it like this for the moment. A revision of content and pedagogy is another (noble) task.

Reply:
I checked the version of Numba consequent on `environment.yml`:
![image](https://github.com/esciencecenter-digital-skills/parallel-python-workbench/assets/21219611/b971ef90-3850-4b6b-afed-db37243c9e40)
It seems then too early to assume that the default is `nopython=True` (v 0.59.0 as per the deprecation warning). 

The Python in my course environment is 3.11.3. Apologies for having revised the code snippets today with Python 3.8 instead of 3.11. Just my bad.